python/ray/tune/trainable.py

```diff
@@ -544,6 +546,11 @@ def restore(self, checkpoint_path: str, checkpoint_node_ip: Optional[str] = None
                 # Only keep for backwards compatibility
                 self.storage_client.sync_down(external_uri, local_dir)
                 self.storage_client.wait_or_retry()
+            elif os.path.exists(checkpoint_path):
+                try:
+                    TrainableUtil.find_checkpoint_dir(checkpoint_path)
+                except Exception:
+                    pass
```

Comment:
I think we can move this check earlier. Basically

```
checkpoint_exists = False

if os.path.exists(checkpoint_path):
    try:
        TrainableUtil.find_checkpoint_dir(checkpoint_path) 
    except Exception:
        pass
    else:
        checkpoint_exists = True

if checkpoint_exists:
    pass
elif self.uses_cloud_checkpointing:
    # ...
```

then we'll never download if the checkpoint already exists.

Reply:
agreed, made the change